### ISSUE TITLE
TypeError with dsolve and a system of ODEs

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#14001]
See system from #8193.  Now this raises different exception:
```
In [1]: V, I = symbols("V I", cls=Function)

In [2]: RC, t, C, Vs, L, R1, V0, I0 = symbols("RC t C Vs L R1 V0 I0")

In [3]: system = [Eq(V(t).diff(t), -1/RC*V(t) + I(t)/C), Eq(I(t).diff(t), -R1/L*I(t) - 1/L*V(t) + Vs/L)]

In [4]: ics = {V(0): V0, I(0): I0}

In [5]: dsolve(system, [V(t), I(t)], ics=ics)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
...
~/src/sympy/sympy/core/relational.py in __nonzero__(self)
    227 
    228     def __nonzero__(self):
--> 229         raise TypeError("cannot determine truth value of Relational")
    230 
    231     __bool__ = __nonzero__

TypeError: cannot determine truth value of Relational
```

I hope, second time this will not be "solved" in same way...

**Comment by normalhuman** [comment]
Support for ODE systems with symbolic coefficients is forthcoming #13812

**Comment by normalhuman** [comment]
Without the ics, the solution according to #13812 would be 
```
[Eq(V(t), RC*Vs/(C*R1 + RC) + Piecewise((C1*exp(t*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L)))/C + C2*exp(t*(-sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L)))/C, Ne((L - R1*RC)**2/(L**2*RC**2) - 4/(C*L), 0)), 
((C2*(t/C + 1/(C*(-sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) + R1/(2*L)))) + C1/C)*exp(t*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L))), True))), 
Eq(I(t), C*Vs/(C*R1 + RC) + Piecewise((C1*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 + 1/(2*RC) - R1/(2*L))*exp(t*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L))) + C2*(-sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 + 1/(2*RC) - R1/(2*L))*exp(t*(-sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L))), Ne((L - R1*RC)**2/(L**2*RC**2) - 4/(C*L), 0)), 
((C1*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 + 1/(2*RC) - R1/(2*L)) + C2*t*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 + 1/(2*RC) - R1/(2*L)))*exp(t*(sqrt((-1/RC + R1/L)**2 - 4/(C*L))/2 - 1/(2*RC) - R1/(2*L))), True)))]
```

which `solve` can't handle, apparently because of this Piecewise. There is room for improvement in `solve` to better handle equations with Piecewise, but this is a separate issue. 

My point here is that  although #13812 does not resolve this issue completely, it does the part that the ODE module is supposed to do; the rest is on `solve`.

**Comment by mijo2** [comment]
@oscarbenjamin This issue can be closed now as, after the merging of PR #19341, this issue gets solved.

**Comment by oscarbenjamin** [comment]
It can be closed when the regression test is merged
